provider/oci/provider.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package oci
+
+import (
+	"github.com/juju/errors"
+	"github.com/juju/jsonschema"
+	"github.com/juju/loggo"
+	"github.com/juju/schema"
+	"gopkg.in/juju/environschema.v1"
+
+	"github.com/juju/juju/cloud"
+	"github.com/juju/juju/environs"
+	"github.com/juju/juju/environs/config"
+)
+
+var logger = loggo.GetLogger("juju.provider.oracle")
+
+// EnvironProvider type implements environs.EnvironProvider interface
+type EnvironProvider struct{}
+
+var _ config.ConfigSchemaSource = (*EnvironProvider)(nil)
+var _ environs.ProviderSchema = (*EnvironProvider)(nil)
+
+// Schema implements environs.ProviderSchema
+func (o *EnvironProvider) Schema() environschema.Fields {
+	return nil
+}
+
+// ConfigSchema implements config.ConfigSchemaSource
+func (o *EnvironProvider) ConfigSchema() schema.Fields {
+	return nil
+}
+
+// ConfigDefaults implements config.ConfigSchemaSource
+func (o *EnvironProvider) ConfigDefaults() schema.Defaults {
+	return nil
+}
+
+// Version implements environs.EnvironProvider.
+func (e EnvironProvider) Version() int {
+	return 0
```

Comment:
I would prefer if we start versions at 1, so that you can tell a difference from the 'zero value' of an int.